moveit_commander/src/moveit_commander/planning_scene_interface.py

```diff
@@ -167,23 +109,23 @@ def add_plane(self, name, pose, normal = (0, 0, 1), offset = 0):
         p.coef.append(offset)
         co.planes = [p]
         co.plane_poses = [pose.pose]
-        self._pub_co.publish(co)
-        
-    def attach_mesh(self, link, name, pose = None, filename = '', size = (1, 1, 1), touch_links = []):
+        self.__submit(co, attach=False)
+
+    def attach_mesh(self, link, name, pose=None, filename='', size=(1, 1, 1), touch_links=[]):
         aco = AttachedCollisionObject()
-        if pose!=None and filename:
+        if (pose is not None) and filename:
             aco.object = self.__make_mesh(name, pose, filename, size)
         else:
             aco.object = self.__make_existing(name)
         aco.link_name = link
         aco.touch_links = [link]
         if len(touch_links) > 0:
             aco.touch_links = touch_links
-        self._pub_aco.publish(aco)
+        self.__submit(aco, attach=True)
 
-    def attach_box(self, link, name, pose = None, size = (1, 1, 1), touch_links = []):
+    def attach_box(self, link, name, pose=None, size=(1, 1, 1), touch_links=[]):
```

Comment:
Also, this function could definitely have a docstring explaining what each parameter does. Some of the behavior is not intuitive. (eg: Why does pose -> make new object rather than move existing object)

Reply:
I don't know. I've only refactored this part of the code to complain with PEP8, none of it's contents are from me. 

`attach_box()` is used in the moveit commander interpreter (I'm not sure how much use that gets) while `attach_mesh()` remains unused within the moveit codebase.

I would opt for removing both and moving this to a follow-up issue to consolidate functionality. That logic is indeed not intuitive with the current interface, I think the `add_()` interfaces should expose the `attach` flag and we should provide a separate set of methods which operate *only* on existing objects. This way we could rid of the logic around `pose is not None` implying intent.

Reply:
sounds good. Maybe add a TODO if you have a moment